modelon/impact/client/experiment_definition.py

```diff
@@ -366,6 +395,303 @@ def to_dict(self):
+
+        fmi_target --
+            Compiler target. Possible values are 'me' and 'cs'. Default: 'me'.
+
+        fmi_version --
+            The FMI version. Valid options are '1.0' and '2.0'. Default: '2.0'.
+
+        platform --
+            Platform for FMU binary. Supported values are "auto", "win64", "win32"
+            or "linux64". Default: 'auto'.
+
+        compiler_log_level --
+            The logging for the compiler. Possible values are "error",
+            "warning", "info", "verbose" and "debug". Default: 'warning'.
+
+        runtime_options --
+            An runtime options class instance of
+            modelon.impact.client.options.ExecutionOptions or
+            a dictionary object containing the runtime options. Default: None.
+
+        solver_options --
+            The solver options to use for this experiment. By default the options
+            is set to None, which means the default options for the
+            custom_function input is used.
+
+        simulation_options --
+            The simulation_options to use for this experiment. By default the
+            options is set to None, which means the default options for the
+            custom_function input is used.
+
+        simulation_log_level --
+            Simulation log level for this experiment. Default is 'WARNING'.
+
+    Examples::
+
+        model = workspace.get_model("Modelica.Blocks.Examples.PID_Controller")
+        simulation_options = custom_function.get_simulation_options()
+        .with_values(ncp=500)
+        solver_options = {'atol':1e-8}
+        simulate_def = model.new_experiment_definition(custom_function,
+        solver_options, simulation_options)
+        simulate_def.to_dict()
+    """
+
+    def __init__(
+        self,
+        model,
+        custom_function,
+        compiler_options=None,
+        fmi_target="me",
+        fmi_version="2.0",
+        platform="auto",
+        compiler_log_level="warning",
+        runtime_options=None,
+        solver_options=None,
+        simulation_options=None,
+        simulation_log_level="WARNING",
+    ):
+        assert_valid_args(
+            model=model,
+            custom_function=custom_function,
+            compiler_options=compiler_options,
+            runtime_options=runtime_options,
+            solver_options=solver_options,
+            simulation_options=simulation_options,
+        )
+        self.model = model
+        self.custom_function = custom_function
+        self._compiler_options = (
+            dict(compiler_options)
+            if isinstance(compiler_options, ExecutionOptions)
+            else compiler_options
+            if isinstance(compiler_options, dict)
+            else {}
+        )
+        self._runtime_options = (
+            dict(runtime_options)
+            if isinstance(runtime_options, ExecutionOptions)
+            else runtime_options
+            if isinstance(runtime_options, dict)
+            else {}
+        )
```

Comment:
These options should also get default options from the custom function if they are None. After that this construct will be repeated 6 times in this file and some refactoring should be done. Also make sure the documentation is updated to mention what None means.